### PR TITLE
Soft migration to Generalize Multinode client for EVM BCI-2286

### PR DESCRIPTION
**Pull request by dhaidashenko**
* Created feature flag that injects EVM client that is based on common/client/MultiNode

### CODE REVIEW

File: core/chains/evm/chain.go [hunk 1]
```diff
@@ -213,15 +214,24 @@ func NewTOMLChain(ctx context.Context, chain *toml.EVMConfig, opts ChainRelayExt
 	return newChain(ctx, cfg, chain.Nodes, opts)
 }
 
-func newChain(ctx context.Context, cfg *evmconfig.ChainScoped, nodes []*toml.Node, opts ChainRelayExtenderConfig) (*chain, error) {
+func buildClient(cfg *evmconfig.ChainScoped, nodes []*toml.Node, opts ChainRelayExtenderConfig) (evmclient.Client, error) {
 	chainID, chainType := cfg.EVM().ChainID(), cfg.EVM().ChainType()
 	l := opts.Logger
+	if cfg.Feature().EVMClientV2() {
```
Comment (dimriou) [inline core/chains/evm/chain.go:220]: Can you add a Debug message so we can easily confirm with NOPs if it's indeed enabled?

File: core/config/docs/core.toml [hunk 2]
```diff
@@ -13,6 +13,8 @@ FeedsManager = true # Default
 LogPoller = false # Default
 # UICSAKeys enables CSA Keys in the UI.
 UICSAKeys = false # Default
+# EVMClientV2 enables experimental RPC client for EVM chains that is based on Generalized MultiNode component.
```
Comment (dimriou) [inline core/config/docs/core.toml:16]: IMO `experimental` is used in POC components.
```suggestion
# EVMClientV2 enables a generalized MultiNode client for EVM chains.
```

File: docs/CONFIG.md [hunk 3]
```diff
@@ -72,6 +73,12 @@ UICSAKeys = false # Default
 ```
 UICSAKeys enables CSA Keys in the UI.
 
+### EVMClientV2
+```toml
+EVMClientV2 = false # Default
+```
+EVMClientV2 enables experimental RPC client for EVM chains that is based on Generalized MultiNode component.
```
Comment (dimriou) [inline docs/CONFIG.md:80]: Same comment as above

**Review by dimriou — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

File: core/chains/legacyevm/chain.go [hunk 4]
```diff
@@ -474,46 +471,27 @@ func (c *chain) Logger() logger.Logger                    { return c.logger }
 func (c *chain) BalanceMonitor() monitor.BalanceMonitor   { return c.balanceMonitor }
 func (c *chain) GasEstimator() gas.EvmFeeEstimator        { return c.gasEstimator }
 
-func newEthClientFromChain(cfg evmconfig.NodePool, noNewHeadsThreshold time.Duration, lggr logger.Logger, chainID *big.Int, chainType commonconfig.ChainType, nodes []*toml.Node) (evmclient.Client, error) {
-	var primaries []evmclient.Node
-	var sendonlys []evmclient.SendOnlyNode
+func newEthClientFromCfg(cfg evmconfig.NodePool, noNewHeadsThreshold time.Duration, lggr logger.Logger, chainID *big.Int, chainType commonconfig.ChainType, nodes []*toml.Node) evmclient.Client {
```
Comment (dimriou) [inline core/chains/legacyevm/chain.go:474]: Is there a reason why we renamed this method?

**Review by jmank88 — APPROVED** [review]

File: core/chains/legacyevm/chain.go (hunk 4 above)
Comment (dimriou) [reply core/chains/legacyevm/chain.go:474]: Ok, sounds reasonable

**Review by dimriou — APPROVED** [review]

**Pull request merged**